polkadot/node/core/pvf/prepare-worker/src/lib.rs

```diff
@@ -119,20 +117,57 @@ async fn send_response(stream: &mut UnixStream, result: PrepareResult) -> io::Re
 /// 7. Send the result of preparation back to the host. If any error occurred in the above steps, we
 ///    send that in the `PrepareResult`.
 pub fn worker_entrypoint(
-	socket_path: &str,
+	worker_dir_path: PathBuf,
 	node_version: Option<&str>,
 	worker_version: Option<&str>,
+	security_status: SecurityStatus,
 ) {
 	worker_event_loop(
 		"prepare",
-		socket_path,
+		worker_dir_path,
 		node_version,
 		worker_version,
-		|mut stream| async move {
+		&security_status,
+		|mut stream, worker_dir_path| async move {
 			let worker_pid = std::process::id();
+			let temp_artifact_dest = worker_dir::prepare_tmp_artifact(&worker_dir_path);
+
+			// Try to enable landlock.
+			{
+				#[cfg(target_os = "linux")]
+				let landlock_status = {
+					use polkadot_node_core_pvf_common::worker::security::landlock::{
+						path_beneath_rules, try_restrict, AccessFs,
+					};
+
+					// Allow an exception for writing to the known file in the worker cache.
+					try_restrict(path_beneath_rules(&[&temp_artifact_dest], AccessFs::WriteFile))
+						.map(LandlockStatus::from_ruleset_status)
+						.map_err(|e| e.to_string())
+				};
+				#[cfg(not(target_os = "linux"))]
+				let landlock_status: Result<LandlockStatus, String> = Ok(LandlockStatus::NotEnforced);
+
+				// Error if the host determined that landlock is fully enabled and we couldn't fully
+				// enforce it here.
+				if security_status.can_enable_landlock &&
+					!matches!(landlock_status, Ok(LandlockStatus::FullyEnforced))
+				{
+					gum::warn!(
+						target: LOG_TARGET,
+						%worker_pid,
+						"could not fully enable landlock: {:?}",
+						landlock_status
+					);
+					return Err(io::Error::new(
```

Comment:
Yeah. We already checked for it so I can't see how this could fail. But I didn't feel confident enough to make it an assert. Makes total sense to leave it as a warning right now.